proto/state/v1alpha1/state.proto

```diff
@@ -16,30 +16,40 @@ syntax = "proto3";
 
 package state.v1alpha1;
 
+import "buf/validate/validate.proto";
+
 // GlobalState is a sorted array managed modules, each one with the latest
 // reference from its local array of references. This is kept updated in a
 // global state file at the root sync directory.
 message GlobalState {
+  option (buf.validate.message).cel = {
+    id: "module_state.unique_module_names",
+    expression: "size(this.modules.filter(reference, !this.modules.map(i, i.module_name).exists_one(x, x == reference.module_name))) == 0 ? '' : (this.modules.map(reference, reference.module_name).filter(module_name, !this.modules.map(reference, reference.module_name).exists_one(x, x == module_name)))[0] + ' has appeared multiple times'"
```

Comment:
Thx to @oliversun9

Reply:
Can you document what this expression does? It's difficult to parse, I _think_ it's checking that `modules.module_name` is unique? If so, you could clarify the error message: 

> module with name %s has appeared multiple times

Reply:
Improved the error message (check the tests), and documented the CEL expression.